cocos/core/assets/texture-2d.ts

```diff
@@ -270,21 +270,6 @@ export class Texture2D extends SimpleTexture {
         return Object.assign(texInfo, presumed);
     }
 
-    protected _checkTextureLoaded () {
```

Comment:
If we ignore the ImageAsset upload and make texture directly usable, then its size will be 0 before ImageAsset downloaded. That could cause potential issue.

A way to resolve this is to serialize and deserialize size for ImageAsset and Texture2D, which we haven't done yet

Reply:
The reason currently you haven't seen any issue, is because we have serialized size in SpriteFrame.
But it's a structural issue in the Asset design, in native asset manager for future version, we will have multithread asset loading process, that means asynchronous loading.